kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportMapper.kt

```diff
@@ -75,8 +75,15 @@ private fun isSealedClassConstructor(descriptor: ConstructorDescriptor) = descri
 
 // Note: partially duplicated in ObjCExportLazyImpl.translateTopLevels.
 internal fun ObjCExportMapper.shouldBeExposed(descriptor: CallableMemberDescriptor): Boolean =
-        descriptor.isEffectivelyPublicApi && !descriptor.isExpect &&
-                !isHiddenByDeprecation(descriptor) && !(descriptor is ConstructorDescriptor && isSealedClassConstructor(descriptor))
+        descriptor.isEffectivelyPublicApi && !descriptor.isExpect && !isHiddenByDeprecation(descriptor) &&
+                !(descriptor is ConstructorDescriptor && isSealedClassConstructor(descriptor)) && !descriptor.isRefinedForObjC()
+
+private fun CallableMemberDescriptor.isRefinedForObjC(): Boolean = when {
+    overriddenDescriptors.isNotEmpty() -> overriddenDescriptors.all { it.isRefinedForObjC() }
```

Comment:
This is a bit suboptimal. Instead of checking all overridden declarations, we can simply check a singe one (the frontend checker prohibits all cases where this would lead to different results, right?).

Reply:
Correct, both `isRefinedForObjC` and `isRefinedInSwift` will now only check the first overridden descriptor.